designcompose/src/main/java/com/android/designcompose/squoosh/SquooshLayout.kt

```diff
@@ -218,3 +218,11 @@ internal fun layoutTree(
     layoutValueCache.putAll(updatedLayouts)
     populateComputedLayout(root, layoutValueCache)
 }
+
+/// Go over a tree that layout can't be applied to, and copy the
+/// layout values from whatever the source tree is.
+internal fun updateDerivedLayout(n: SquooshResolvedNode) {
+    if (n.layoutNode != null) n.computedLayout = n.layoutNode.computedLayout
+    if (n.firstChild != null) updateDerivedLayout(n.firstChild!!)
```

Comment:
I left it how it was because I have a few other lines of 'if not null', and I couldn't make them so clear (so I felt it was better to be clunky but similar from line to line...)